src/open_prime_hunters_rando/string_tables/string_tables.py

```diff
@@ -0,0 +1,208 @@
+)
+
+
+def num_bytes_to_align(length: int, modulus: int = 4) -> int:
+    if length % modulus > 0:
+        return modulus - (length % modulus)
+    return 0
+
+
+class StringTableAdapter(construct.Adapter):
+    def __init__(self) -> None:
+        super().__init__(StringTableConstruct)
+
+    def _decode(self, obj: Container, context: Container, path: str) -> Container:
+        decoded = copy.deepcopy(obj)
+
+        # wrap strings
+        decoded.strings = ListContainer([StringEntry(string) for string in decoded.strings])
+
+        return decoded
+
+    def _encode(self, obj: Container, context: Container, path: str) -> Container:
+        encoded = copy.deepcopy(obj)
+
+        strings = typing.cast("list[StringEntry]", encoded.strings)
+
+        # update sizes and offsets
+        encoded.strings = ListContainer()
+
+        offset = 8 if StringTableHeader.unk is not None else 4
+        offset += RawStringEntry.sizeof() * len(strings)
+
+        for string_wrapper in strings:
+            string = string_wrapper._raw
+
+            size = len(string_wrapper.text)
+            string._string_length = size
+            string._data_offset = offset
+
+            offset += size + num_bytes_to_align(size)
+
+            encoded.strings.append(string)
+
+        return encoded
+
+
+class StringEntry:
+    def __init__(self, raw: Container) -> None:
+        self._raw = raw
+
+    def __repr__(self) -> str:
+        return f"<String string_id={self.string_id}> text={self.text}"
+
+    def __eq__(self, value: Any) -> bool:
+        if not isinstance(value, StringEntry):
+            return False
+        if value.string_id != self.string_id:
+            return False
+
+        def check_container(container: dict, other: dict) -> bool:
+            for k in container.keys() | other.keys():
+                if k.startswith("_"):
+                    continue
+                if isinstance(container[k], dict):
+                    if not isinstance(other[k], dict):
+                        return False
+                    if not check_container(container[k], other[k]):
+                        return False
+                else:
+                    if container[k] != other[k]:
+                        return False
+            return True
+
+        return check_container(self._raw, value._raw)
+
+    @property
+    def string_id(self) -> str:
+        return self._raw.string_id
+
+    @string_id.setter
+    def string_id(self, value: str) -> None:
+        self._raw.string_id = value
+
+    @property
+    def scan_icon(self) -> ScanIcon:
+        return self._raw.scan_icon
+
+    @scan_icon.setter
+    def scan_icon(self, value: ScanIcon) -> None:
+        self._raw.scan_icon = value
+
+    @property
+    def text(self) -> str:
+        return self._raw.text
+
+    @text.setter
+    def text(self, value: str) -> None:
+        self._raw.text = value
+
+    @property
+    def string_length(self) -> int:
```

Comment:
this property doesn't need to exist. just use `len(string_entry.text)`

Reply:
I hope this this what you meant.

Reply:
yeah I would do the latter